src/cpp/cpp_constructor.cpp

```diff
@@ -234,11 +228,8 @@ optionalt<codet> cpp_typecheckt::cpp_constructor(
       cpp_namet(constructor_name, source_location).as_expr();
     function_call.arguments().reserve(operands_tc.size());
 
-    for(exprt::operandst::iterator
-        it=operands_tc.begin();
-        it!=operands_tc.end();
-        it++)
-      function_call.op1().copy_to_operands(*it);
+    for(auto &op : operands_tc)
+      function_call.op1().copy_to_operands(op);
```

Comment:
I think we can get rid of this loop entirely if using a 3-argument constructor for `function_call` above.